client/src/cordova/apple/OutlineAppleLib/Sources/OutlineCatalystApp/CatalystApp.swift

```diff
@@ -53,6 +57,48 @@
             { _ in
                 appKitController._AppKitBridge_setConnectionStatus(.disconnected)
             }
+            
+            // Handle connection toggle from tray menu
+            NotificationCenter.default.addObserver(forName: NSNotification.Name("toggleConnection"),
+                                                   object: nil,
+                                                   queue: nil)
+            { _ in
+                Task {
+                    // First, ensure the window is open
+                    NotificationCenter.default.post(name: NSNotification.Name("openApplication"), object: nil)
+                    
+                    // Wait a bit for the window to open
+                    try? await Task.sleep(nanoseconds: 500_000_000) // 0.5 seconds
```

Comment:
This is a race condition. We should not rely on timing.

But this whole taske should be removed. See below.